aggregator/src/app/EthereumService.ts

```diff
@@ -53,73 +53,99 @@ export default class WalletService {
+    const wallet = EthereumService.Wallet(aggPrivateKey);
+    const nextNonce = (await wallet.getTransactionCount());
+    const chainId = await wallet.getChainId();
     const blsWalletSigner = await initBlsWalletSigner({ chainId });
 
-    return new WalletService(
+    return new EthereumService(
       emit,
-      aggregatorSigner,
+      wallet,
       blsWalletSigner,
       verificationGatewayAddress,
       nextNonce,
     );
   }
 
-  async checkTx(tx: TransactionData): Promise<TxCheckResult> {
-    const signedCorrectly = this.blsWalletSigner.verify(tx);
+  async BlockNumber(): Promise<BigNumber> {
+    return BigNumber.from(
+      await this.wallet.provider.getBlockNumber(),
+    );
+  }
 
-    const failures: TransactionFailure[] = [];
+  async waitForNextBlock() {
+    await new Promise((resolve) => {
+      this.wallet.provider.once("block", resolve);
+    });
+  }
 
-    if (signedCorrectly === false) {
-      failures.push({
-        type: "invalid-signature",
-        description: "invalid signature",
-      });
-    }
+  // TODO (merge-ok): Consider: We may want to fail operations
+  // that are not at the next expected nonce, including all
+  // current pending transactions for that wallet.
+  async checkNonces(bundle: Bundle): Promise<TransactionFailure[]> {
+    const failures: TransactionFailure[] = [];
 
-    const nextNonce = await BlsWallet.Nonce(
-      tx.publicKey,
-      this.verificationGateway.address,
-      this.aggregatorSigner,
+    // Not our responsibility to check these are consistent, but we do want to
+    // ensure that our indexes are valid.
+    const len = Math.min(
+      bundle.operations.length,
+      bundle.senderPublicKeys.length,
     );
 
-    if (BigNumber.from(tx.nonce).lt(nextNonce)) {
-      failures.push({
-        type: "duplicate-nonce",
-        description: [
-          `nonce ${tx.nonce} has already been processed (the next nonce for`,
-          `this wallet will be ${nextNonce.toString()})`,
-        ].join(" "),
-      });
+    for (let i = 0; i < len; i++) {
+      const nextNonce = await BlsWalletWrapper.Nonce(
+        bundle.senderPublicKeys[i],
+        this.verificationGateway.address,
+        this.wallet,
+      );
+
+      if (nextNonce.gt(bundle.operations[i].nonce)) {
+        failures.push({
+          type: "duplicate-nonce",
+          description: [
+            `operation ${i}: nonce ${bundle.operations[i].nonce} has already`,
+            `been processed (the next nonce for this wallet will be`,
+            `${nextNonce.toString()})`,
+          ].join(" "),
+        });
+      }
     }
 
-    return { failures, nextNonce };
+    return failures;
+  }
+
+  async checkBundle(bundle: Bundle) {
+    try {
+      const { successes } = await this.verificationGateway.callStatic
+        .processBundle(bundle);
+      // All operations in the bundle should be estimated to succeed.
+      return successes.every((suc) => suc);
+    } catch {
+      return false;
```

Comment:
This will silently fail on random errors sch as RPC issues. Should we log something here so error is no swallowed? We may also want to eventually bubble that error up one level so the error can be attached to the bundle row in case a bundle submitter wants to learn why their operations are not being submitted.